### PR TITLE
fixed prefix to mention with at sign

### PR DESCRIPTION
**Pull request by jung1411**
https://github.com/dhmoon91/discord/issues/3

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ node_modules
 df_styled.png
 .env
 **/__pycache__
+last_match.html
+df_style.css
```
Comment (dhmoon91) [inline .gitignore]: what are these two files that you are ignoring?

Comment (dhmoon91) [inline .gitignore]: what are these two files that you are ignoring?

File: riot.py [hunk 2]
```diff
@@ -53,12 +53,14 @@ def previousMatch(name:str):
     participants_row['Deaths'] = row['stats']['deaths']
     participants_row['Assists'] = row['stats']['assists']
     participants_row['Damage'] = row['stats']['totalDamageDealt']
-    participants_row['Gold'] = row['stats']['goldEarned']
+    # participants_row['Gold'] = row['stats']['goldEarned']
     # participants_row['champLevel'] = row['stats']['champLevel']
-    participants_row['Minions'] = row['stats']['totalMinionsKilled']
+    # participants_row['Minions'] = row['stats']['totalMinionsKilled']
```
Comment (dhmoon91) [inline riot.py]: why comment?

File: .gitignore (hunk 1 above)
Comment (jung1411) [reply .gitignore]: 아아 일단 저거 두개 나중에 html  image 바꿔서 그거 시도해볼려고 나두긴했는데 다른데다가 저장하고 지우고 올리는게 낫나?

File: riot.py (hunk 2 above)
Comment (jung1411) [reply riot.py]: 일단 이것저것 시도해보는중에 커멘팅함

File: .gitignore (hunk 1 above)
Comment (dhmoon91) [reply .gitignore]: `html image 바꿔서` 이게 몬소리임?

Comment (jung1411) [reply .gitignore]: 어제 dataframe을 table 형식으로 바꿨는데 css입히고 그거를 image로 convert해서 bot한테 send하는과정에서 html을 image로 convert하는 과정에서 계속 이상하게 변형돼서 일단 지금은 다시 dataframe으로 아웃풋으로 바꿈 근데 나중에 시간날때 고쳐볼려고.

File: riot.py [hunk 3]
```diff
@@ -16,10 +16,39 @@ def getSummonerRank(name: str):
   user = watcher.summoner.by_name(my_region, name)
   ranked_stat= watcher.league.by_summoner(my_region, user['id'])
 
+  # Get summoner Icon Image
+  profileiconid = user['profileIconId']
+  version = watcher.data_dragon.versions_for_region(my_region)['v']
+  summoner_icon_image_url = f"http://ddragon.leagueoflegends.com/cdn/{version}/img/profileicon/{profileiconid}.png"
```
Comment (dhmoon91) [inline riot.py]: why using url like this? you can't get it through api?

File: riot.py [hunk 4]
```diff
@@ -16,10 +16,39 @@ def getSummonerRank(name: str):
...
   # Find solo queue data.
   solo_rank_stat =  pydash.find(ranked_stat, {"queueType":"RANKED_SOLO_5x5"})
-  
-  return ' '.join([solo_rank_stat['tier'], solo_rank_stat['rank']])
+
+  tier_division = solo_rank_stat['tier']
+  tier_rank = solo_rank_stat['rank']
+
+
+  tier_image = f"ranked-emblems/Emblem_{tier_division}.png"
```
Comment (dhmoon91) [inline riot.py]: You don't have this image from api?

**Comment by dhmoon91** [comment]
You are putting two stuff in this PR again. 
One for fixing prefix and other one for formatting the output.

The prefix stuff looks good but this PR can't be merged cause you put other stuff in this PR.

File: .gitignore (hunk 1 above)
Comment (dhmoon91) [reply .gitignore]: I have no idea what you are talkng about... What are these two files used for??

File: riot.py (hunk 3 above)
Comment (dhmoon91) [reply riot.py]: okay looks like not

File: riot.py (hunk 2 above)
Comment (dhmoon91) [reply riot.py]: actually this is unrelated changes. Revert back please

File: bot.py [hunk 5]
```diff
@@ -19,7 +19,7 @@
 TOKEN = os.getenv('DISCORD_TOKEN')
 
 # TODO: Update prefix to be called with bot name.
-bot = commands.Bot(command_prefix="!", intents=intents)
+bot = commands.Bot(command_prefix=commands.when_mentioned_or("!"), intents=intents)
```
Comment (dhmoon91) [inline bot.py]: Why do we need `_or` ? This will take commands using `!` as well right?? I don't think we wnat that

File: bot.py [hunk 6]
```diff
@@ -36,8 +36,37 @@ async def on_member_join(member):
...
+  summoner_name = summoner_info['user_name']
+  
+  # Removing space of the summoner name to access op.gg url of the summoner 
+  summoner_name_opgg = summoner_name.replace(" ", "")
+  # Add author, thumbnail, fields, and footer to the embed
+  embed.set_author(name=summoner_name, url=f"https://na.op.gg/summoner/userName={summoner_name_opgg}", icon_url=summoner_info['summoner_icon_image_url'])
+  
+  # Get image of tier by path
+  tier_image = summoner_info['tier_image']
+  file = discord.File(tier_image)
```
Comment (dhmoon91) [inline bot.py]: What does this do?

File: bot.py (hunk 5 above)
Comment (jung1411) [reply bot.py]: 근데 일단 deploy하기전에 빼면돼는데 매번 커맨드 칠때 @관전남-nj 하는것보다 !가 더 편해서
!command 할때는 그냥 복붙하면돼는데 @관전남-nj는 매번 쳐야돼서

File: bot.py (hunk 6 above)
Comment (jung1411) [reply bot.py]: 저게 없으면 thumbnail로 image를 send 못시켜서 embed.set_thumbnail할때 인식을못해

Comment (jung1411) [reply bot.py]: tier_image path 불러와서 discord에다가 file object를 보낼때

File: .gitignore (hunk 1 above)
Comment (jung1411) [reply .gitignore]: 원래 html으로 send할때 필요했는데 안돼서 일단 안쓰고있음 지금으로선 필요없음

File: bot.py (hunk 5 above)
Comment (dhmoon91) [reply bot.py]: Code never should change between deployments. 
Maybe we can use env var? what happens if you put empty string?

File: .gitignore (hunk 1 above)
Comment (dhmoon91) [reply .gitignore]: okay then remove it please

File: bot.py (hunk 5 above)
Comment (jung1411) [reply bot.py]: empty string해도 그대로 적용됌 (rank leonaldu) 이렇게 해도 실행돼

File: riot.py (hunk 4 above)
Comment (jung1411) [reply riot.py]: 아 찾아보니깐 있다. 이것도 고칠께

File: bot.py (hunk 5 above)
Comment (dhmoon91) [reply bot.py]: okay we don't want that then. 
Put this as env variable. 
check how we are getting env var for the tokens. 
Locally we can set it to `!` and on prod we can set it to crazy string like `aosidhaosiheoigheogieh3434234234owehfgofiuehge` so it won't listen to it.

Update readme as well please

**Comment by dhmoon91** [comment]
@jung1411 fix the conflict please
Also remove the image files if you are not using it anymore

File: riot.py (hunk 4 above)
Comment (dhmoon91) [reply riot.py]: this fixed?

File: .gitignore (hunk 1 above)
Comment (jung1411) [reply .gitignore]: removed

**Comment by dhmoon91** [comment]
did you remove the image files under emblem/ or whatever? I still see the files in PR

File: riot.py (hunk 4 above)
Comment (jung1411) [reply riot.py]: 아아 tier icon은 api 안에 없음
어제 새벽에 잠결에 보느라 profile icon 말한걸로 착각함

File: bot.py [hunk 7]
```diff
@@ -17,17 +17,19 @@
...
 
 @bot.event
 async def on_ready():
+    """Prints that the bot is connected"""
     print(f"{bot.user.name} has connected to Discord!")
 
 
 @bot.event
 async def on_member_join(member):
+    """Sends personal discord message to the membed who join"""
```
Comment (dhmoon91) [inline bot.py:37]: what is this triple """?

File: bot.py (hunk 5 above)
Comment (dhmoon91) [reply bot.py]: this is still not fixed?

**Comment by jung1411** [comment]
fixed all error except import for pylint checks

**Review by dhmoon91 — APPROVED** [review]

**Pull request merged**